src/backend/gpopt/CGPOptimizer.cpp

```diff
@@ -245,14 +257,34 @@ InitGPOPT()
 {
 	GPOS_TRY
 	{
-		return CGPOptimizer::InitGPOPT();
+		CGPOptimizer::InitGPOPT();
 	}
 	GPOS_CATCH_EX(ex)
 	{
 		if (GPOS_MATCH_EX(ex, gpdxl::ExmaGPDB, gpdxl::ExmiGPDBError))
 		{
 			PG_RE_THROW();
 		}
+
+		GPOS_TRY
+		{
+			CGPOptimizer::TerminateGPOPT(true);
+		}
+		GPOS_CATCH_EX(ex)
```

Comment:
I suggest to use different names for the CException variables to easily distinguish them.

Reply:
done